hooks/shapediver/viewer/interaction/gumball/useGumball.ts

```diff
@@ -0,0 +1,174 @@
+ * @param gumballProps Parameter properties to be used. This includes name filters, and properties for the behavior of the gumball.
+ * @param activate Set this to true to activate the gumball. If false, preparations are made but no gumball is possible.
+ * @param initialTransformedNodeNames The initial transformed node names (used to initialize the selection state).
+ * 					Note that this initial state is not checked against the filter pattern.
+ */
+export function useGumball(
+	sessionId: string, 
+	viewportId: string, 
+	gumballProps: IGumballParameterProps,
+	activate: boolean,
+	initialTransformedNodeNames?: { name: string, transformation: number[] }[]
+): IGumballState {
+	// get the session API
+	const sessionApi = useShapeDiverStoreViewer(state => { return state.sessions[sessionId]; });
+	// get the viewport API
+	const viewportApi = useShapeDiverStoreViewer(state => { return state.viewports[viewportId]; });
+	
+	// create the selection settings from the interaction settings
+	const selectionSettings = useMemo(() => {
+		if (!gumballProps) return {};
+
+		return {
+			nameFilter: gumballProps.nameFilter,
+			hover: gumballProps.hover,
+			minimumSelection: 0,
+			maximumSelection: Infinity
+		} as ISelectionParameterProps;
+	}, [gumballProps]);
+
+	// use the selection hook to get the selected node names
+	const { selectedNodeNames, setSelectedNodeNames } = useSelection(sessionId, viewportId, selectionSettings, activate);
+	// use the gumball events hook to get the transformed node names
+	const { transformedNodeNames, setTransformedNodeNames } = useGumballEvents(selectedNodeNames, initialTransformedNodeNames);
+
+	// create a reference for the gumball
+	const gumballRef = useRef<Gumball | undefined>(undefined);
+
+	// use an effect to create the gumball whenever the selected node names change
+	useEffect(() => {
+		if (viewportApi) {
+			// whenever this output node changes, we want to create the interaction engine
+			const nodes = getNodesByName(sessionApi, selectedNodeNames);
+			const gumball = new Gumball(viewportApi, Object.values(nodes).map(n => n.node));
+			gumballRef.current = gumball;
+		}
+
+		return () => {
+			// clean up the select manager
+			if (gumballRef.current) {
+				gumballRef.current.close();
+				gumballRef.current = undefined;
+			}
+		};
+	}, [viewportApi, sessionApi, selectedNodeNames]);
+	
+	/**
+	 * Restore the transformed node names.
+	 * 
+	 * This function is used to restore the transformed nodes to their new transformation state.
+	 * This means that the transformation of the nodes is updated to the new transformation state.
+	 * 
+	 * @param newTransformedNodeNames The new transformed node names.
+	 * @param oldTransformedNodeNames The old transformed node names.
+	 * @returns
+	 */
+	const restoreTransformedNodeNames = useCallback((newTransformedNodeNames: { name: string, transformation: number[] }[], oldTransformedNodeNames: { name: string, transformation: number[] }[]) => {
```

Comment:
@MajorMeerkatThe3rd detail: we could remove the transformation property from the type definition for `oldTransformedNodeNames`